python/ray/autoscaler/command_runner.py

```diff
@@ -536,7 +536,7 @@ def run(
             ssh_options_override_ssh_key="",
     ):
         if run_env == "auto":
-            run_env = "host" if cmd.find("docker") == 0 else "docker"
+            run_env = "docker" if (cmd.find("docker") > -1 and cmd.find("docker pull") == -1) else "host"
```

Comment:
Why would we want to run `docker commands` inside of the container?

Reply:
The previous commit was indeed running docker commands inside the container. cmd.find("docker") returns 0 if "docker" is found in cmd at index 0, which is not what we want

Reply:
The code is working as intended. The previous logic was to find commands like `docker ps` or `docker container ls`, all which need to be run outside the container. The wrapping of commands with the `docker exec -it <container_name>...` happens below.